src/main/java/gyro/google/storage/BucketAccessControlResource.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.storage;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.model.BucketAccessControl;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidStrings;
+import gyro.google.Copyable;
+import gyro.google.GoogleResource;
+
+import java.io.IOException;
+import java.util.Set;
+
+/**
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::bucket 'acl-bucket'
+ *         name: 'acl-example'
+ *         location: 'US-CENTRAL1'
+ *     end
+ *
+ *     google::acl 'acl-one'
+ *        bucket: $(google::bucket acl-bucket)
+ *        entity: 'domain-brightspot.com'
+ *        role: 'OWNER'
+ *     end
+ *
+ *     google::acl 'acl-two'
+ *        bucket: $(google::bucket acl-bucket)
+ *        entity: 'domain-sample.com'
+ *        role: 'READER'
+ *     end
+ */
+@Type("acl")
+public class BucketAccessControlResource extends GoogleResource implements Copyable<BucketAccessControl> {
+    
+    private static final String ENTITY_REGEX = "\\b(allUsers|allAuthenticatedUsers)\\b|^(user|group|domain|project)-.*?";
+
+    private BucketResource bucket;
+    private String role;
+    private String entity;
+    private String userProject;
+    private String id;
+    private String selfLink;
+    private String domain;
+    private String email;
+    private String entityId;
+    private BucketAccessControlProjectTeam projectTeam;
```

Comment:
Are all these output fields, if so their getters shout have the `@Output` annotation.

Reply:
Fixed.